### ISSUE TITLE
Make some variables, objects and data global and rethink models.

### ISSUE THREAD

**Issue by Slacki** [gernest/utron#51]
# I am missing access to some data and objects when writing an application with utron.

I would like to provide these three examples, so you can understand what I mean. I will try to show how inaccessible database and configuration, in places other than controller action, affect the code transparency and make **M**odel less important than it should be.

## Example 1

Let's say there is some go routine running in background taking care of something. The go routine uses database. To access DB **somewhere else** than in controller action I had to do this dirty hack (the function is a middleware running on each request, which is not efficient at all)

```
var DB *utron.Model

func dbGlobal(c *utron.Context) error {
	c.DB.LogMode(true)
	if DB == nil {
		DB = c.DB
	}

	return nil
}
```

Then finally I can access DB in go routine like this

```
func thisIsGoRoutine() {
	// wait for first request so middleware is executed and db is accessible
	for DB == nil {
		time.Sleep(100 * time.Millisecond)
	}

	// do some stuff
}
```

## Example 2

There are some components I have to initialize and pass some parameters to them. They are scattered all around the project. I can't store these parameters in config file, because I can't access it (again) anywhere else than in controller action (via context).

```
var SessionStorage, _ = mysqlstore.NewMySQLStore(
	"root:@tcp(127.0.0.1:3306)/argaroth_automigrate?charset=utf8&parseTime=True&loc=Local",
	"sessions",                  // db table name
	"/",                         // cookie path
	60*60*24*30,                 // expire time
	[]byte(SESSION_STORAGE_KEY), // secret key for cookie
)
```

### Example 3

I want to write some so called `business logic` in models in order to keep controllers tidy and transparent. Then again I can't do that without having global DB access done like in example 1.

## In conclusion

At the moment Model part of MVC is reduced to struct definition. In my opinion it shouldn't be that way.

## Discussion

I would like to hear what do you think about what i've said. Maybe I am missing something here, then I am open to learn something new.
